### PR TITLE
Add release notes for v2.1.0-rc.1

### PR DESCRIPTION
**Pull request by jseldess**
See https://github.com/cockroachdb/cockroach/issues/31359.

### CODE REVIEW

File: releases/v2.1.0-rc.1.md [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+- Fixed schema change rollback caused by GC TTL threshold error. [#31153][#31153]
+- CockroachDB now supports CTEs inside [views](../v2.1/views.html). [#31051][#31051]
+- CockroachDB now hints that internal errors should be [reported as bugs by users](../v2.1/file-an-issue.html). Additionally, internal errors are now collected internally and submitted (anonymized) with other node statistics when statistic collection is enabled. [#31272][#31272]
+- It is now possible to force a specific index for `DELETE` or `UPDATE`. [#31279][#31279]
+- Handle binary fields dumped by `mysqldump v5.7.23` with `_binary` prefix. [#31305][#31305]
+- `EXPLAIN ANALYZE <statement>` is now a valid equivalent of [`EXPLAIN ANALYZE (DISTSQL) <statement>`](../v2.1/explain-analyze.html) [#31278][#31278]
+- CockroachDB now collects references to tables in [`information_schema`](../v2.1/information-schema.html) and `pg_catalog` that are not yet implemented and report them as telemetry if statistics reporting is enabled. This will help determine which features should be implemented next for compatibility. [#31357][#31357]
```
Comment (rmloveland) [inline releases/v2.1.0-rc.1.md]: should be plural: "and reports them"

File: releases/v2.1.0-rc.1.md [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+
+- The **Service latency: {90,99}th percentile** graphs on the [**Overview**](../v2.1/admin-ui-overview-dashboard.html) and [**SQL**](../v2.1/admin-ui-sql-dashboard.html) dashboards, as well as the P50 and P99 latency numbers in the time series area sidebar, now reflect latencies of both local and distributed queries. Previously, they only included local queries. [#31116][#31116]
+- Links to documentation pages now open in a new tab. [#31132][#31132]
+- Improved the view of databases with no tables. [#31231][#31231]
+- Updated [**Jobs**](../v2.1/admin-ui-jobs-page.html) dashboard to make each row expandable, allowing the user to see the error message for failed jobs. [#31237][#31237]
+
+### Bug fixes
+
+- Fixed the `_admin/v1/enqueue_range` debug endpoint to always respect its `node_id` parameter. [#31087][#31087]
+- CockroachDB now reports an unimplemented error when a common table expression containing [`INSERT`](../v2.1/insert.html)/[`UPDATE`]((../v2.1/update.html)/[`UPSERT`](../v2.1/upsert.html)/[`DELETE`](../v2.1/delete.html) is not otherwise used in the remainder of the query. [#31051][#31051]
```
Comment (rmloveland) [inline releases/v2.1.0-rc.1.md]: `UPDATE` link not rendering properly due to `((`, I think

File: releases/v2.1.0-rc.1.md [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+- CockroachDB does not silently ignore `WITH` clauses within parentheses anymore. [#31051][#31051]
+- Fixed a rare scenario where a [backup](../v2.1/backup.html) could incorrectly include a key for an aborted transaction. [#31316][#31316]
+- CockroachDB now avoids repeatedly trying a replica that was found to be in the process of being added. [#31250][#31250]
+- CockroachDB will no longer fail in unexpected ways or write invalid data when the type of input values provided to [`INSERT`](../v2.1/insert.html)/[`UPSERT`](../v2.1/upsert.html) does not match the type of the target columns. [#31280][#31280]
+- [`UPDATE`](../v2.1/update.html) now verifies the column constraints before [`CHECK`](../v2.1/check.html) constraints, for compatibility with PostgreSQL. [#31280][#31280]
+- It is no longer possible to use not-fully-added-yet columns in the `RETURNING` clause of [`UPDATE`](../v2.1/update.html) statements. [#31280][#31280]
+- CockroachDB no longer incorrectly and silently accepts a [computed column](../v2.1/computed-columns.html) on the left side of the assignment in a [`ON CONFLICT DO UPDATE SET`](../v2.1/insert.html#on-conflict-clause) clause. [#31280][#31280]
```
Comment (rmloveland) [inline releases/v2.1.0-rc.1.md]: I think this and the next list item should be "in an `ON CONFLICT ...` clause".

**Comment by lhirata** [comment]
TFTRs! @RaduBerinde - ready to merge whenever binaries are ready.

**Comment by RaduBerinde** [comment]
I'm waiting on github to kick off the final verification but it's not working yet (they're still working on their issues).

**Comment by RaduBerinde** [comment]
@lhirata the binaries are ready (the linux one is https://binaries.cockroachdb.com/cockroach-v2.1.0-rc.1.linux-amd64.tgz)

**Pull request merged**